xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
It seems you can get rid of readLock variable and include the information inside nameSpaceLock itself so you won't need to specify readLock again in xl.unlockNS(volume, path)